authors/apps/articles/views.py

```diff
@@ -139,3 +137,43 @@ def put(self, request, pk):
                 {
                     'error': 'You cannot edit an article you do not own. '
                 }, status.HTTP_403_FORBIDDEN)
+
+
+class FavoriteArticle(generics.ListCreateAPIView):
+    """
+    A user is able to favourite an article if had not favourited it.
+    If they had favourited it, thearticle becomes unfavourited.
+    """
+    permission_classes = (IsAuthenticated, )
+    queryset = Article.objects.all()
+    serializer_class = ArticleSerializer
+
+    def post(self, request, slug):
+        """
+        Checks whether the article exists.
+        Checks whether the user has favourited the article in order to favourite
+        it or unfavourite it if the user had already favourited it.
+        """
+        try:
+            article = Article.objects.get(slug=slug)
+        except Exception:
```

Comment:
I suggest you change this exception to ObjectDoesNotExist since its the one you are looking for at this point. You should not catch general exceptions, rather catch specific ones.